### PR TITLE
Flask practice homework

### PR DESCRIPTION
**Pull request by vadiq**

### CODE REVIEW

File: flask_lesson_intro/main.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@app.route('/battery-charger')
+def get_page_battery_charger():
+    return render_template("battery-charger.html", title='Battery charger', data=get_data())
+
+
+@app.route('/author')
+def get_page_author():
+    return render_template("author.html", title='Author')
```
Comment (dimigor) [inline flask_lesson_intro/main.py:24]: try to make universal route

File: flask_lesson_intro/templates/author.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{% extends 'layout.html' %}
+
+{% block content %}
+<h1>Vadim Savchuk</h1>
+<div><a href="https://github.com/vadiq/common">>>> GitHub repo <<<</a></div>
+<div>I did it without looking to the previous homework, but with little looking to my notes :)</div>
+{% endblock content %}
```
Comment (dimigor) [inline flask_lesson_intro/templates/author.html]: missed endline and tabulation

File: flask_lesson_intro/templates/home.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{% extends 'layout.html' %}
+
+{% block content %}
+<div id="images">
+    <img src="{{url_for('static', filename='flask-homework.png')}}" usemap="#image-map" width="650px" height="360px">
+    <map name="image-map">
+        <area href="/alarm-clock" coords="20,20,167,132" shape="rect">
+        <area href="/headphones" coords="392,143,242,11" shape="rect">
```
Comment (dimigor) [inline flask_lesson_intro/templates/home.html]: use `url_for` instead

File: flask_lesson_intro/templates/layout.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    <meta charset="UTF-8">
+    {% if title %}
+    <title>Site test - {{ title }}</title>
+    {% else %}
+    <title>Site test</title>
+    {% endif %}
+    <!-- Bootstrap core CSS -->
+    <link href="https://getbootstrap.com/docs/4.3/dist/css/bootstrap.min.css" rel="stylesheet"
+          crossorigin="anonymous">
+    <style>
```
Comment (dimigor) [inline flask_lesson_intro/templates/layout.html]: for styles create file `styles.css`

File: flask_lesson_intro/utils.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import json
+
+
+def get_data():
+    with open("data.json") as file:
+        return json.load(file)
```
Comment (dimigor) [inline flask_lesson_intro/utils.py:6]: what if file will not found?

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_lesson_intro/utils.py (hunk 5 above)
Comment (vadiq) [reply flask_lesson_intro/utils.py:6]: if file will not be found - then some of my app pages will be lacking of some info

File: flask_lesson_intro/templates/layout-items.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+{% block content %}
+<h1>{{ item.upper() }}</h1>
+<div>
+    <div>
+        <p><img width="40%" src="{{url_for('static', filename=item + '.png')}}"></p>
+    </div>
+    <div>
+        {% for element in data %}
+        {% if element.title == item %}
```
Comment (dimigor) [inline flask_lesson_intro/templates/layout-items.html]: maybe you should do this logic at the route?

**Review by dimigor — CHANGES_REQUESTED** [review]

**Review by dimigor — APPROVED** [review]

**Pull request merged**